File: action-type/15-chain-of-responsibility/go/porn_filter.go
package main

import (
	"fmt"
	"regexp"
)

// PornFilter 涉黄涉暴过滤器

type PornFilter struct{}

func (f *PornFilter) Filter(text string, chain *FilterChain) string {

	result := regexp.MustCompile(`黄色|暴力`).ReplaceAllString(text, "[敏感词]")
	fmt.Println("[PornFilter] 处理后:", result)

	return chain.Filter(result)
}

File: action-type/15-chain-of-responsibility/java/PornFilter.java
package main;

// 涉黄涉暴过滤器
public class PornFilter implements SensitiveFilter {
   
    @Override
    public String doFilter(String text, FilterChain chain) {
        
        String result = text.replaceAll("黄色|暴力", "[敏感词]");
        System.out.println("[PornFilter] 处理后: " + result);
   
        return chain.doFilter(result);
    }
}
